mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/TaskExecutorRegistration.java

```diff
@@ -141,4 +147,35 @@ public Optional<String> getAttributeByKey(String attributeKey) {
 
         return Optional.empty();
     }
+
+    /**
+     * Computes the scheduling constraints based on the properties of the current registration.
+     *
+     * The `SizeDefinition` is created using the `machineDefinition` and by extracting the `sizeName`
+     * from the registration's attributes, if it exists.
+     *
+     * The `allocationAttributes` are also extracted from the registration's attributes. The method analyzes all entries within the provided
+     * `allocationAttributesWithDefault`. When a key is found within the registration's attributes,
+     * its corresponding value is used. If a key is not found, it uses the associated
+     * default value from `allocationAttributesWithDefault`.
+     *
+     * @param allocationAttributesWithDefault A Map where keys are resource allocation attributes required for scheduling,
+     *                                        and values are default options used when specific attributes are not present in the registration's attributes.
+     *
+     * @return A `SchedulingConstraints` object, built from the `SizeDefinition` and the processed `allocationAttributes`.
+     */
+    @JsonIgnore
+    public SchedulingConstraints getSchedulingConstraints(Map<String, String> allocationAttributesWithDefault) {
```

Comment:
I'm wondering why a Task Executor (TE) would have a method that returns its scheduling constraints. Wouldn't it be better to have a Fitness Calculator that takes into account both the Scheduling Constraints of the task that needs to be scheduled and the Task Executor metadata (TE Registration in this case) and then tries to find how good the match is?